packages/core/src/shared/handleUninstall.ts

```diff
@@ -7,59 +7,113 @@
 
 import * as vscode from 'vscode'
 import { existsSync } from 'fs'
+import * as semver from 'semver'
 import { join } from 'path'
 import { getLogger } from './logger/logger'
 import { telemetry } from './telemetry'
 import { VSCODE_EXTENSION_ID } from './extensions'
 import { extensionVersion } from './vscode/env'
 
 /**
- * Checks if the extension has been uninstalled by reading the .obsolete file
- * and comparing the number of obsolete extensions with the installed extensions.
+ * Checks if an extension has been uninstalled and performs a callback if so.
+ * This function differentiates between an uninstall and an auto-update.
  *
- * @param {string} extensionName - The name of the extension.
- * @param {string} extensionsDirPath - The path to the extensions directory.
- * @param {string} obsoleteFilePath - The path to the .obsolete file.
- * @param {function} callback - Action performed when extension is uninstalled.
- * @returns {void}
+ * @param extensionId - The ID of the extension to check (e.g., VSCODE_EXTENSION_ID.awstoolkit)
+ * @param extensionsPath - The file system path to the VS Code extensions directory
+ * @param obsoletePath - The file system path to the .obsolete file
+ * @param onUninstallCallback - A callback function to execute if the extension is uninstalled
  */
 async function checkExtensionUninstall(
-    extensionName: typeof VSCODE_EXTENSION_ID.awstoolkit | typeof VSCODE_EXTENSION_ID.amazonq,
-    extensionsDirPath: string,
-    obsoleteFilePath: string,
-    callback: () => Promise<void>
+    extensionId: typeof VSCODE_EXTENSION_ID.awstoolkit | typeof VSCODE_EXTENSION_ID.amazonq,
+    extensionsPath: string,
+    obsoletePath: string,
+    onUninstallCallback: () => Promise<void>
 ): Promise<void> {
-    /**
-     * Users can have multiple profiles with different versions of the extensions.
-     *
-     * This makes sure the callback is triggered only when an explicit extension with specific version is uninstalled.
-     */
-    const extension = `${extensionName}-${extensionVersion}`
+    const extensionFullName = `${extensionId}-${extensionVersion}`
```

Comment:
It seems inconsistent to pull `extensionVersion` from the environment, but you have to pass in the extension ID. The caller has the choice of what ID to pass but in reality since the version will only match to the running extension there is not actually a choice.
 Why not pass in both or pull both from the env, e.g. `vscode.ExtensionContext.extension.id`.